### ISSUE TITLE
LFS support available?

### ISSUE THREAD

**Issue by fanktom** [SwiftGit2/SwiftGit2#152]
Hey there,

is there any option or recommendation on how to use this library with LFS?

I know lfs itself implements this with clean/smudge filters - which could also work with libgit2.
Or is the idea to wait for libgit2 to integrate it and then support it?

**Comment by mdiep** [comment]
The intention of this library was to be Swift bindings to libgit2. That doesn't mean that that's the best approach, but that was the intention. 🙂

**Comment by tiennou** [comment]
The "expectation" is that AFAICT everything you need should be available, only the wiring is missing. In other words, `libgit2` has (some) [user-specified filtering support](https://github.com/libgit2/libgit2/blob/HEAD/include/git2/sys/filter.h), which IIRC is wired to be done automatically on checkout, so you "just need" to write the callbacks ? The thing is that (for now), there's a policy of not forking in the library (because it's hard to do correctly), so if Git-LFS needs that you have to handle it in your application code.

In short, if you manage to make it work, it's likely there's interest in having some of that support code upstream.

**Comment by fanktom** [comment]
Thanks for the insights @tiennou, I will have a look at it and report any progress!
